app/src/main/java/org/fcitx/fcitx5/android/input/popup/PopupComponent.kt

```diff
@@ -78,25 +79,79 @@ class PopupComponent :
         }
     }
 
-    private fun showPopup(viewId: Int, content: String, bounds: Rect) {
+    private fun showPopup(
+        viewId: Int,
+        content: String,
+        bounds: Rect,
+        style: PopupAction.PreviewStyle
+    ) {
         showingEntryUi[viewId]?.apply {
             dismissJobs[viewId]?.also {
                 dismissJobs.remove(viewId)?.cancel()
             }
             lastShowTime = System.currentTimeMillis()
-            setText(content)
+            // Always clear armed state when (re)showing a preview
```

Comment:
Should have create a new [`PopupContainerUi`](https://github.com/fcitx5-android/fcitx5-android/blob/d769ff1b0cb22dc465f2cd7a456b017dfbe8325d/app/src/main/java/org/fcitx/fcitx5/android/input/popup/PopupContainerUi.kt#L15) subclass instead of cluttering everywhere